tests/integration/conftest.py

```diff
@@ -107,13 +112,19 @@ def _get_plugin_list_from_ip(unit_ip):
     return _get_plugin_list_from_ip
 
 
-@pytest.fixture
-def openstack_environment(request):
+@pytest.fixture(scope="module", name="openstack_environment")
+def openstack_environment_fixture(request, num_units):
     """Parse the openstack rc style configuration file from the --openstack-rc argument
 
-    Return a dictionary of environment variables and values.
+    Returns: a dictionary of environment variables and values, or None if --openstack-rc isn't
+        provided.
     """
     rc_file = request.config.getoption("--openstack-rc")
+    assert (
+        num_units == 1 or rc_file
+    ), "swift plugin is required for multi-unit deployment, please include an openstack configuration in the --openstack-rc parameter "
```

Comment:
This line looks like it is longer than 99 characters?

Reply:
Black does not break the string literal even if it's longer than 99 characters.

Reply:
ah yes, you have to do it manually in those cases. `pylint` should complain though, unless `line-too-long` is disabled 😉  which it is in this repo

Reply:
Fixed, thanks!